### ISSUE TITLE
Test modules trigger `inline-modules` when linting all targets

### ISSUE THREAD

**Issue by git-staus** [rust-lang/rust-clippy#17342]
### Summary

When Clippy is ran on all targets, even test modules make the inline module lint fire, even though the [documentation](https://rust-lang.github.io/rust-clippy/master/index.html#inline_modules) explicitly says modules annotated with `#[cfg(test)]` would be ignored.

I believe linting test code is a legitimate use case, as it ensures the test code is held to the same standard as the code it is supposed to represent.

### Lint Name

clippy::inline-modules

### Reproducer

I tried this code:

```rust
fn main() {}
#[cfg(test)]
mod tests {}
```

I invoked Clippy like so:

```bash
cargo clippy --all-targets -- -Wclippy::inline-modules
```

I saw this happen:

```
warning: inline module found
 --> src/main.rs:3:1
  |
3 | mod tests {}
  | ^^^^^^^^^^^^
  |
  = help: move the contents of the module to `src/tests.rs` or `src/tests/mod.rs`, and replace this with `mod tests;`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#inline_modules
  = note: requested on the command line with `-W clippy::inline-modules`

warning: `issue_17342` (bin "issue_17342" test) generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.01s
```

I expected to see this happen:

```
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.01s
```

### Version

```text
rustc 1.98.0-nightly (4c9d2bfe4 2026-07-01)
binary: rustc
commit-hash: 4c9d2bfe4ad7a65669098754964aaebe0ec1ced2
commit-date: 2026-07-01
host: x86_64-unknown-linux-gnu
release: 1.98.0-nightly
LLVM version: 22.1.8
```
